folding-schemes/Cargo.toml

```diff
@@ -23,9 +23,13 @@ num-integer = "0.1"
 color-eyre = "=0.6.2"
 ark-bn254 = {version="0.4.0"}
 ark-groth16 = { version = "^0.4.0" }
+ark-noname = { git = "https://github.com/dmpierre/ark-noname", branch="feat/sonobe-integration" }
+noname = { git = "https://github.com/dmpierre/noname" }
+serde_json = "1.0.85"                                                                # to (de)serialize JSON
```

Comment:
It's weird to need the specific serde variant as a direct dep instead of serde itself only and the variant as dev-dep.

Can you double-check that is it needed?

Reply:
I am using `serde` to [derive the `Serialize, Deserialize` traits](https://github.com/privacy-scaling-explorations/sonobe/blob/d8e8f205296cca064edc943ce082d58aebe575ac/folding-schemes/src/frontend/noname/utils.rs#L11) and `serde_json` to [handle serializing and deserializing structs and json data](https://github.com/privacy-scaling-explorations/sonobe/blob/d8e8f205296cca064edc943ce082d58aebe575ac/folding-schemes/src/frontend/noname/mod.rs#L104). If I understand correctly your suggestions, wouldn't it be the other way around? Specifying `serde` as a dev dep and `serde_json` as a direct dependency?

Reply:
Normally is in the other way. But in this case JSON is the only accepted format. So sadly we need to import on this way. 

It's a bit weird IMO. As we could have bincode and yaml too but they force JSON for whatever reason..